File: go/hamming/hamming.go
package hamming

import "errors"

// Distance returns the number of differences between the two strands
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("strands must be the same length")
	}

	count := 0
	for i := range a {
		if a[i] != b[i] {
			count++
		}
	}

	return count, nil
}

File: ruby/hamming/hamming.rb
# A class for computing the number of differences in two strings
# (Hamming distance)
class Hamming
  def self.compute(first_string, second_string)
    raise ArgumentError if first_string.length != second_string.length

    (0...first_string.length).count { |i| first_string[i] != second_string[i] }
  end
end

module BookKeeping
  VERSION = 3
end